static/js/compose_validate.js

```diff
@@ -654,6 +655,81 @@ function validate_private_message() {
     return true;
 }
 
+export function check_send_permissions() {
+    const send_button = document.querySelector("#compose-send-button");
+    const send_button_container = document.querySelector(".compose_right_float_container");
+    const tippy_instance = send_button_container._tippy || tippy(send_button_container);
+    const msg_type = compose_state.get_message_type();
+
+    if (msg_type === "private" && page_params.realm_private_message_policy === 2) {
+        const recipients = compose_pm_pill.get_user_ids();
+
+        const non_bot_users = recipients.filter(
+            (user_id) => !people.get_by_user_id(user_id).is_bot,
+        );
+
+        if (non_bot_users.length > 0) {
+            tippy_instance.setContent(
+                $t_html({
+                    defaultMessage:
+                        "You are not allowed to send private messages in this organization.",
+                }),
+            );
+            send_button.classList.add("button_disabled");
+            send_button.disabled = true;
+            return;
+        }
+    }
+
+    if (msg_type === "stream") {
+        const stream_name = $("#stream_message_recipient_stream").val();
+        const stream = stream_data.get_sub(stream_name);
+
+        if (!stream) {
+            return;
+        }
+
+        const post_policy = stream.stream_post_policy;
+        const post_policy_values = stream_data.stream_post_policy_values;
+        const {is_admin, is_moderator, realm_waiting_period_threshold, user_id} = page_params;
+        const {admins, moderators, non_new_members} = post_policy_values;
+
+        const tooltip_content = $t_html(
+            {defaultMessage: "You are not allowed to send messages to #{stream_name}."},
+            {stream_name},
+        );
+
+        if (post_policy === admins.code && !is_admin) {
+            tippy_instance.setContent(tooltip_content);
+            send_button.disabled = true;
+            return;
+        }
+        if (post_policy === moderators.code && !is_admin && !is_moderator) {
+            tippy_instance.setContent(tooltip_content);
+            send_button.disabled = true;
+            return;
+        }
```

Comment:
If we want to show same content in tooltip for all the cases, we can extract a function to check whether a user is allowed to post or not.

Reply:
As per your suggestion, I have extracted out the logic for stream posting permissions and moved it into a separate function